Assets/Scripts/Fish/Fish.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Fish : MonoBehaviour
+{
+    // stamina / recovery (longterm and shortterm)
+
+    [SerializeField]
+    private float drag;
+
+    private Kinematic kinematic;
+    private Vector3 externalVelocity;
+    private Vector3 momentum;
+    private Vector3 angularMomentum;
+
+    private void Awake() => kinematic = new Kinematic();
+
+    void Update()
+    {
+        HandlePhysics();
+
+        // Calculate transform with kinematic linear movement + any other force applied
+        transform.position = transform.position + (kinematic.LinearVel * Time.deltaTime) + (momentum * Time.deltaTime);
+        transform.rotation = kinematic.Orientation;
```

Comment:
Currently, the Flock script takes care of applying velocities on its Flock Agents. I suppose we could do a refactoring so Flocks rely instead on the Fish script 🤔 I'll look into that once you've merged this PR.